File: Modules/Common/AbnormalInfo/SIE.Web.AbnormalInfo/AbnormalMonitors/Commands/WritingRuleCommand.js
SIE.defineCommand('SIE.Web.AbnormalInfo.AbnormalMonitors.Commands.WritingRuleCommand', {
    extend: 'SIE.cmd.Edit',
    meta: { text: "编写规则", group: "edit", iconCls: "icon-ClipboardVariantEdit icon-blue" },
    canExecute: function (view) {
        return view.getCurrent();
    },
    execute: function (view, source) {
        var me = this;
        var indata = {};
        var editEntity = this.getEditEntity();
        indata.Data = Ext.encode(editEntity.data);
        view.execute({
            data: indata,
            success: function (res) {
                if (res.Success) {
                    me.onEditting(editEntity);
                    me.edit(editEntity);
                    me.onEdited(editEntity);
                    me.showView(editEntity);
                }
            },
            error: function (res) {
                Ext.Msg.alert('提示', res.Message);
            }
        });
    },
    //弹出页签效果
    showView: function (editEntity) {
        var me = this;
        CRT.Workbench.addPage({
            entityType:"SIE.AbnormalInfo.AbnormalMonitors.AbnormalDecisionRule",
            recordId: editEntity.getId(),
            title: me.getEditViewTitle(editEntity),
            viewGroup: "WritingRule",
            isDetail: true,
            ignoreQuery: true,
            params: {
                Id: editEntity.getId(),
                MonitorName: editEntity.getMonitorName(),
                MonitorType: editEntity.getMonitorType(),
                MonitorTabName: editEntity.getMonitorTabName()
            },
            pageClass: 'SIE.Web.AbnormalInfo.AnomalyMonitors.WritingRuleViewPage'
        });
    }
});

File: Modules/Common/AbnormalInfo/SIE.Web.AbnormalInfo/AbnormalMonitors/Commands/WritingRuleCommand.cs
using SIE.AbnormalInfo.AbnormalMonitors;
using SIE.Web.Command;
using System;

namespace SIE.Web.AbnormalInfo.AbnormalMonitors.Commands
{
	/// <summary>
	/// 来料检验填写报告命令
	/// </summary>
	[JsCommand("SIE.Web.AbnormalInfo.AbnormalMonitors.Commands.WritingRuleCommand")]
    public class WritingRuleCommand : ViewCommand<ViewArgs>
    {
        /// <summary>
        /// 执行方法
        /// </summary>
        /// <param name="args">视图参数</param>
        /// <param name="scope">SIE.MetaModel.View.Block.EntityType</param>
        /// <returns>object</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var bill = args.Data.ToJsonObject<AbnormalDecisionRule>();
            if (null == bill)
            {
                throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(bill)));
            }

            //var iqcBill = RT.Service.Resolve<IqcBillController>().WritingReport(bill.Id);
            return bill;
        }
    }
}
